src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx

```diff
@@ -0,0 +1,283 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import space from 'app/styles/space';
+import {t} from 'app/locale';
+import TextField from 'app/components/forms/textField';
+import TextOverflow from 'app/components/textOverflow';
+import Tooltip from 'app/components/tooltip';
+import {defined} from 'app/utils';
+
+import {
+  selectors,
+  valueSelectors,
+  booleanSelectors,
+  Suggestion,
+  Suggestions,
+} from './dataPrivacyRulesPanelSelectorFieldTypes';
+
+type State = {
+  showOptions: boolean;
+  suggestions: Suggestions;
+  fieldValues: Array<Suggestion | string>;
+};
+
+type Props = {
+  value: string;
+  onChange: (value: string) => void;
+  error?: string;
+  onBlur?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
+  disabled?: boolean;
+};
+
+class DataPrivacyRulesPanelSelectorField extends React.Component<Props, State> {
+  state: State = {
+    suggestions: [],
+    fieldValues: [],
+    showOptions: false,
+  };
+
+  componentWillMount() {
+    document.addEventListener('mousedown', this.handleClickOutside, false);
+  }
+
+  componentDidMount() {
+    this.loadFieldValues(this.props.value);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.handleClickOutside, false);
+  }
+
+  selectorField = React.createRef<HTMLDivElement>();
+
+  loadFieldValues = (newValue: string) => {
+    const splittedValue = newValue.split(' ');
+    const fieldValues: Array<Suggestion | string> = [];
+
+    for (const index in splittedValue) {
+      const value = splittedValue[index];
+      const selector = selectors.find(s => s.value === value);
+      fieldValues.push(selector ? selector : value);
+    }
+
+    const newSuggestions = this.getNewSuggestions(fieldValues);
+
+    this.setState({
+      fieldValues,
+      suggestions: newSuggestions,
+    });
+  };
+
+  getNewSuggestions = (fieldValues: Array<Suggestion | string>) => {
+    const lastFieldValue = fieldValues[fieldValues.length - 1];
+    const penultimateFieldValue = fieldValues[fieldValues.length - 2];
+
+    if (!defined(lastFieldValue)) {
+      return [];
+    }
+
+    if (
+      typeof penultimateFieldValue === 'object' &&
+      penultimateFieldValue.type === 'boolean'
```

Comment:
Can we do something like `penultimateFieldValue?.type === 'boolean'` without checking if it's of type object first?

Reply:
not really, but I will check if I can improve this part of the code